rosbag2_transport/test/rosbag2_transport/test_record_services.cpp

```diff
@@ -152,3 +169,25 @@ TEST_F(RecordSrvsTest, trigger_snapshot)
   successful_service_request<Snapshot>(cli_snapshot_);
   EXPECT_THAT(mock_writer.get_messages().size(), Ne(0u));
 }
+
+class RecordSrvsSplitBagfileTest : public RecordSrvsTest
+{
+protected:
+  RecordSrvsSplitBagfileTest()
+  : RecordSrvsTest(false /*snapshot_mode*/) {}
+};
+
+TEST_F(RecordSrvsSplitBagfileTest, split_bagfile)
+{
+  auto & writer = recorder_->get_writer_handle();
+  MockSequentialWriter & mock_writer =
+    static_cast<MockSequentialWriter &>(writer.get_implementation_handle());
+  EXPECT_FALSE(mock_writer.split_bagfile_called());
+
+  std::chrono::duration<float> duration(2.0);
+  std::this_thread::sleep_for(duration);
```

Comment:
Please remove this sleep. 
We publishing for 2.5 seconds, it should be enough to receive at least on message needed for split.